### PR TITLE
migrate posts end points with jwt

### PR DESCRIPTION
**Pull request by Karrar-Mohammed**
as in title

### CODE REVIEW

File: src/main/kotlin/com/the_chance/controllers/PostsController.kt [hunk 1]
```diff
@@ -9,9 +9,10 @@ import com.the_chance.utils.NoPostFoundError
 
 
 class PostsController(private val postService: PostService) {
-    suspend fun createPost(content: String?): Post {
+    suspend fun createPost(creatorId: String?, content: String?) {
         if (isValidContent(content)) {
-            return postService.createPost(content!!)
+            val creatorUUID = isValidUUID(creatorId)
```
Comment (AmeerAmjed) [inline src/main/kotlin/com/the_chance/controllers/PostsController.kt:14]: why check validation uuidCreate you already get from token its safe place

File: src/main/kotlin/com/the_chance/data/post/Post.kt [hunk 2]
```diff
@@ -6,5 +6,6 @@ import kotlinx.serialization.Serializable
 data class Post(
     val id: String,
     val content: String,
-    val createdAt: Long
-)
\ No newline at end of file
+    val createdAt: Long,
+    val creatorName: String
```
Comment (AmeerAmjed) [inline src/main/kotlin/com/the_chance/data/post/Post.kt:10]: why just creatorName in post class why should provide object user, in design need name and jobTitle and avatar will need object user when need update or delete this post how check is owner post name not unique

File: src/main/kotlin/com/the_chance/data/post/PostService.kt [hunk 3]
```diff
@@ -17,42 +18,42 @@ class PostService(database: Database) {
...
-            id = newPost[PostTable.id].value.toString(),
-            content = newPost[PostTable.content],
-            createdAt = newPost[PostTable.creationTime]
-        )
     }
 
     suspend fun getAllPost(): List<Post> {
         return dbQuery {
-            PostTable.selectAll().map {
+            (PostTable innerJoin UserTable)
```
Comment (AmeerAmjed) [inline src/main/kotlin/com/the_chance/data/post/PostService.kt:30]: you already here create innerJoin to get info user why use just name need return object user.

**Review by AmeerAmjed — CHANGES_REQUESTED** [review]
good work, need change some point.

**Review by AmeerAmjed — APPROVED** [review]

**Review by mustafa-sab3alarab — APPROVED** [review]

**Pull request merged**